subprojects/core/src/main/java/org/gradle/internal/classpath/InstrumentingTransformer.java

```diff
@@ -347,7 +358,7 @@ public void visitMethodInsn(int opcode, String owner, String name, String descri
                 return;
             }
 
-            if (generatedInterceptor.visitMethodInsn(className, opcode, owner, name, descriptor, isInterface, asNode)) {
+            if (generatedInterceptor.visitMethodInsn(className, opcode, owner, name, descriptor, isInterface, metadata)) {
```

Comment:
(cosmetics) Given that the `isInstanceOf` part does not vary between the calls to `generatedInterceptor.visitMethodInsn`, do you think this could be passed to the `generatedInterceptor`'s constructor instead? This will require separating it from `getMethodNode`, but could arguably improve readability.

Reply:
I like it that `getMethodNode` gets wrapped into an abstraction, though. But it should not necessarily be the same abstraction anyway.

Reply:
I created common abstraction since it's less work when you introduce new method. But I now pass metadata via constructor and getMethodNode is passed as before.